### PR TITLE
Bluespace jars can now contain reagents

### PR DESCRIPTION
**Pull request by jupyterkat**
## About The Pull Request

Ports Citadel-Station-13/Citadel-Station-13#13143

## Why It's Good For The Game

NO NOT THE JAR

image0
## Changelog
:cl:
tweak: bluespace jars can now contain reagents
/:cl:
image1

### CODE REVIEW

**Comment by amidoingitright** [comment]
the fact that you have done this makes me angry

**Comment by Singularity69** [comment]
Finally, I can put the cum reagent into the jar while someone is inside.

**Comment by ZombiesVsPlants** [comment]
O H  N O

**Comment by jbox144** [comment]
every day

we stray further towards becoming a cit downstream

**Comment by park66665** [comment]
Cit rebase when

File: austation/code/game/objects/items/bluespace_jar.dm [hunk 1]
```diff
@@ -221,38 +237,62 @@
...
 
+/obj/item/bluespace_jar/process()
+	if(!reagents)
+		return
+	for(var/mob/living/L in occupants)
+		if(!ishuman(L))
+			continue
+		if((reagents.total_volume >= sipping_level) || ((reagents.total_volume >= sipping_probably) && prob(50))) //sipp
+			reagents.reaction(L, INGEST) //consume
+			reagents.trans_to(L, transfer_rate)
```
Comment (TheFakeElon) [inline austation/code/game/objects/items/bluespace_jar.dm:292]: tying this to prob(50) won't slow it down that much and isn't very efficient. Try making a new proc that has a world.time delay. You could use something like the process_reagents() proc from the pool controller

File: austation/code/game/objects/items/bluespace_jar.dm [hunk 2]
```diff
@@ -67,6 +76,13 @@
 			. += "<span class='notice'>Alt-click to [locked ? "unlock" : "lock"] its [entrance_name].</span>"
 
 /obj/item/bluespace_jar/attack_self(mob/living/user)
+	if(reagents)
+		if(open)
+			spillable = FALSE
+			reagents.flags = NONE
+		else
+			spillable = TRUE
+			reagents.flags = OPENCONTAINER
```
Comment (TheFakeElon) [inline austation/code/game/objects/items/bluespace_jar.dm:85]: just tie the spillable switching to lid opening/closing

**Review by TheFakeElon — CHANGES_REQUESTED** [review]

File: austation/code/game/objects/items/bluespace_jar.dm (hunk 1 above)
Comment (jupyterkat) [reply austation/code/game/objects/items/bluespace_jar.dm:292]: do you know that process() is called every 10 deciseconds?

Comment (jupyterkat) [reply austation/code/game/objects/items/bluespace_jar.dm:292]: the pool fires it every 2 deciseconds even

Comment (TheFakeElon) [reply austation/code/game/objects/items/bluespace_jar.dm:289]: Yes, Process fires every deci second, not once every 20. The pool fires it every 5 deciseconds with its current var, which is more than twice as slow as the jar. Alternatively you could use prob(20) or something similar

Comment (jupyterkat) [reply austation/code/game/objects/items/bluespace_jar.dm:289]: it does fire once every 20 deciseconds, check the subsystem for obj

Comment (jupyterkat) [reply austation/code/game/objects/items/bluespace_jar.dm:289]: pool uses fastprocessing instead of objprocessing

Comment (TheFakeElon) [reply austation/code/game/objects/items/bluespace_jar.dm:289]: 🤔

**Comment by MCterra10** [comment]
slurrrp

**Comment by jbox144** [comment]
this is not ok

**Comment by TheFakeElon** [comment]
We have to draw a line somewhere

**Pull request closed without merging**